enhancements/network/cloud-egress-ip.md

```diff
@@ -0,0 +1,468 @@
+depending on the result. This component will need to execute what the network
+plugin has determined is best. This is done since there is no traffic load
+balancing done today for egress IP intended traffic. The network plugin
+constantly initiates a TCP connection to each egress node and depending on the
+success/failure of that: re-balances the egress assignment to other nodes (thus
+avoiding packets being sent to a egress node which is currently un-reachable).
+Going forward, both network plugins will come to support traffic load-balancing
+and a more sophisticated route health detection. That means that the underlying
+networking solution (OVS in the case of both OVN-Kubernetes and openshift-sdn)
+will manage the route health check. It is worth noting that the "amount of
+re-balancing" done by the network plugin is expected to be reduced, since the
+network fabric will be able to load balance traffic between all egress nodes
+provided and always use a healthy node (thus not requiring the network plugin to
+perform an egress IP re-assignment every time a route goes down).
+
+Given the above, it is however important that the network plugin is conveyed the
+cloud's information concerning the nodes' subnet range. This is especially
+evident on GCP, as without it: the network plugin won't be able to perform any
+assignments at all, given the nodes' default `/32` subnet.
+
+##### OpenShift vs. upstream project
+
+There are two approaches that can be taken for hosting this component. Normally,
+this component would be an OpenShift dedicated component, however since
+ovn-org/ovn-kubernetes does implement the egress IP feature supported on
+bare-metal clusters: it has been discussed whether supporting the cloud egress
+IP feature would make sense for that organization as well.
+
+**upstream**
+
+An "upstream model" would entail getting the sign-off from ovn-org that such a
+feature is useful. Pending a green light, an OpenShift dedicated downstream
+repository would need to be created following that.
+
+The cost of such an approach is more time spent in setting up CI/CD for both
+repositories.
+
+##### CRD vs. a annotated approach
+
+Multiple pieces of information needs to be communicated between the network
+plugins and this cloud component:
+
+- The cloud's node subnet information.
+- The egress IP assignment to be performed by the cloud component
+
+and possibly:
+
+- The status of the egress IP assignment performed by the cloud component
+
+Two models have been imagined for this: a CRD based approach, or a node
+annotated approach.
+
+**CRD**
+
+The cloud component could use a dedicated CRD that the network plugin watches.
+The CRD could store the nodes' subnet, as known by the cloud. The network plugin
+could also use this CRD to issue requests for cloud egress IP assignments, and
+the cloud component could update a status field indicating a successful cloud
+assignment of the egress IP.
```

Comment:
Actually I believe it would be better if the cloud component annotates the node object with the "cloud subnet information", and that the CRD is used by the network plugin to issue a "cloud egress IP request". It feels like splitting the "cloud subnet information" from the "egress IP request" is better.